source/jormungandr/jormungandr/scenarios/helper_classes/helper_utils.py

```diff
@@ -155,118 +180,167 @@ def _extend_journey(pt_journey, fallback_dp, fallback_period_extremity):
-                            dep_mode,
-                        )
-                    ]
-                )
-            else:
-                # extend the journey with the fallback routing path
-                _extend_journey(pt_journey, fallback_dp, fallback_period_extremity)
+class BeginningFallback:
+    def get_pt_section(self, pt_journey):
+        for s in pt_journey.sections:
+            if s.type == response_pb2.CROW_FLY:
+                continue
+            return s
+        return None
+
+    def get_pt_boundaries(self, pt_journey):
+        return self.get_pt_section(pt_journey).origin
+
+    def get_journey_bound_datetime(self, pt_journey):
+        return pt_journey.departure_date_time
+
+    def represent_start(self):
+        return False
+
+    def get_pt_section_datetime(self, pt_journey):
+        return self.get_pt_section(pt_journey).begin_date_time
+
+    def get_fallback_datetime(self, pt_datetime, fallback_duration):
+        return pt_datetime - fallback_duration
+
+    def route_params(self, origin, dest):
+        return origin, dest
+
+    def set_journey_bound_datetime(self, pt_journey):
+        pt_journey.departure_date_time = pt_journey.sections[0].begin_date_time
+
 
+class EndingFallback:
+    def get_pt_section(self, pt_journey):
+        for s in reversed(pt_journey.sections):
+            if s.type == response_pb2.CROW_FLY:
+                continue
+            return s
+        return None
+
+    def get_pt_boundaries(self, pt_journey):
+        return self.get_pt_section(pt_journey).destination
+
+    def get_journey_bound_datetime(self, pt_journey):
+        return pt_journey.arrival_date_time
+
+    def represent_start(self):
+        return True
+
+    def get_pt_section_datetime(self, pt_journey):
+        return self.get_pt_section(pt_journey).end_date_time
+
+    def get_fallback_datetime(self, pt_datetime, fallback_duration):
+        return pt_datetime + fallback_duration
+
+    def route_params(self, origin, dest):
+        return dest, origin
+
+    def set_journey_bound_datetime(self, pt_journey):
+        pt_journey.arrival_date_time = pt_journey.sections[-1].end_date_time
+
+
+def _build_crowflies(pt_journeys, orig, dest):
+    """
+    Update all journeys in pt_journeys with a crowfly as origin and destination fallback
+    """
+    if pt_journeys is None:
+        return
+
+    for pt_journey in pt_journeys.journeys:
+        crowflies = [_build_crowfly(pt_journey, **orig), _build_crowfly(pt_journey, **dest)]
+
+        # Filter out empty crowflies
+        crowflies = [c for c in crowflies if c]
+
+        pt_journey.sections.extend(crowflies)
         pt_journey.sections.sort(SectionSorter())
         pt_journey.departure_date_time = pt_journey.sections[0].begin_date_time
+        pt_journey.arrival_date_time = pt_journey.sections[-1].end_date_time
 
-    return pt_journeys
 
+def _build_crowfly(pt_journey, requested_obj, mode, places_free_access, fallback_durations, fallback_type):
```

Comment:
requested_obj is not very clear for me, how about extremity_obj?

Reply:
What about `entry_point` ? As it seems to be the term used when resolving the origin/dest (see. `get_entry_point_or_raise`)